lib/Doctrine/Common/DataFixtures/Purger/ORMPurger.php

```diff
@@ -96,73 +99,80 @@ public function getObjectManager()
     /** @inheritDoc */
     public function purge()
     {
-        $classes = [];
+        $connection = $this->em->getConnection();
+
+        if ($this->cachedSqlStatements === null) {
+            $this->cachedSqlStatements = [];
+            $classes                   = [];
+
+            foreach ($this->em->getMetadataFactory()->getAllMetadata() as $metadata) {
```

Comment:
instead of wrapping all those statements inside a if, I suggest creating `private function getPurgeStatements: array` to hold all that logic (with an early return if the cache is available)

Reply:
done